src/compile/mark/encode/tooltip.ts

```diff
@@ -116,12 +115,26 @@ export function tooltipRefForEncoding(
     }
   });
 
-  for (const {channel, key, value} of tooltipTuples) {
-    if (!toSkip[channel] && !usedKey[key]) {
-      keyValues.push(`${stringValue(key)}: ${value}`);
-      usedKey[key] = true;
+  const out = {};
+  for (const {channel, key, value} of tuples) {
+    if (!toSkip[channel] && !out[key]) {
+      out[stringValue(key)] = value;
     }
   }
 
+  return out;
+}
+
+export function tooltipRefForEncoding(
+  encoding: Encoding<string>,
+  stack: StackProperties,
+  config: Config,
+  {reactiveGeom}: {reactiveGeom?: boolean} = {}
+) {
+  return tooltipRefData(tooltipData(encoding, stack, config, reactiveGeom));
+}
+
+export function tooltipRefData(data: Dict<string>) {
```

Comment:
Why is this `tooltipRefData`?  This name sounds weird.

I think the two lines here should be just a part of `tooltipRefForEncoding`.

Reply:
Good idea. Done.